app/src/main/java/com/example/tasky/common/model/ResponseHandler.kt

```diff
@@ -0,0 +1,33 @@
+package com.example.tasky.common.model
```

Comment:
With the chosen type of architecture (presentation, domain, data), it's quite unclear what a class in common->model belongs to. Every class can be clearly attributed to one layer, so I'd also put it in the right layer since otherwise you introduce kind of an arbitrary layer like here where no clear rules exist how interaction with it works.

This class handles responses and therefore clearly relates to data. So the proper package would be common->data->util for example (or similar)